source/common/http/http1/codec_impl.cc

```diff
@@ -69,12 +70,16 @@ void StreamEncoderImpl::encodeHeaders(const HeaderMap& headers, bool end_stream)
     saw_content_length = true;
   }
 
+  if (headers.NoChunks()) {
```

Comment:
Can you say a little more about why this is needed? This is a bit odd. I assume it's because you don't want chunk transfer encoding but you don't have a content-length? Is there any way to handle this without this kind of hack?

Reply:
Your assumption is right. I saw that the ":" wes created exactly to allow such kind of hook and it is being used in some other places. Alternatives could be (1) changing the interface of encodeHeaders(), which is much more intrusive since it is all over, or (2) building the response without using encodeHeaders(). What do you recommend?